legacy/src/main/java/macrobase/analysis/transform/BatchScoreFeatureTransform.java

```diff
@@ -39,8 +40,11 @@ public void initialize() {
 
     @Override
     public void consume(List<Datum> records) {
-        if(requiresTraining)
-            batchTrainScore.train(records);
+        if(requiresTraining) {
+            List<Datum> copy = new ArrayList<>(records);
```

Comment:
Not every featuretransform is going to modify its records. Can we add the copy to MCD so other transforms don't need to pay the overhead?

Reply:
Make sense. Will do.